src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs

```diff
@@ -95,23 +102,33 @@ public class WebSocketsTransport : ITransport
                 {
                     using (message)
                     {
-                        await _webSocket.SendAsync(new ArraySegment<byte>(message.Payload.Buffer.ToArray()),
+                        try
+                        {
+                            await _webSocket.SendAsync(new ArraySegment<byte>(message.Payload.Buffer.ToArray()),
                             message.MessageFormat == Format.Text ? WebSocketMessageType.Text : WebSocketMessageType.Binary, true,
                             cancellationToken);
+                        } catch(OperationCanceledException ex)
+                        {
+                            _logger?.LogError("The WebSocket operation was cancelled");
+                        }
                     }
                 }
             }
         }
 
-        public async Task Connect(string url)
+        private async Task Connect(Uri url)
```

Comment:
This is not `Connect` but `prepare url and connect`. I would recommend renaming this method to PrepareUrl/ChangeScheme (or something like this) and just do replacement here and move connect back. Also moving it closer to where it is used will probably make it easier to read.